src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,178 @@
+                'site_url' => $siteUrl,
+            ]);
+            
+            return $command;
+        }
+
+        $request->setSiteUrl($siteUrl);
+        $url = UriTemplateService::expand(
+            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
+        );
+        $request->setInspectionUrl($url);
+        $client = new \Google_Client();
+        $client->addScope(\Google_Service_SearchConsole::WEBMASTERS_READONLY);
+
+        $response = null;
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
+            $client->setAuthConfig(json_decode(base64_decode($base64EncodedAuthConfig), true));
+            $service = new \Google_Service_SearchConsole($client);
+            $response = $service->urlInspection_index->inspect($request);
+        } catch (\Throwable $e) {
+            $this->logger->error('An error occurred in checkIndexStatus for google.', [
+                'exception' => $e,
+                'node_ref' => $node->generateNodeRef(),
+                'url' => $url,
+                'retry_count' => $command->get('ctx_retries'),
+                'stack_trace' => $e->getTraceAsString(),
+            ]);
+        }
+
+        if ($response === null) {
+            return $command;
+        }
+
+        $inspectionResult = $response->getInspectionResult();
+        $indexStatusResult = $inspectionResult->getIndexStatusResult() ?? null;
+        $ampResult = $inspectionResult->getAmpResult() ?? null;
+
+        $isConclusiveForWeb = $indexStatusResult && ($indexStatusResult->getVerdict() === 'PASS' || $indexStatusResult->getVerdict() === 'FAIL');
+        $isConclusiveForAmp = !$node::schema()->hasField('amp_enabled') || ($ampResult && ($ampResult->getVerdict() === 'PASS' || $ampResult->getVerdict() === 'FAIL'));
+
+        if ($isConclusiveForWeb && $isConclusiveForAmp) {
+            $this->putEvent($command, $pbjx, $node, 'google', $response);
+            return $command;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+              
+        if ($retries >= $maxRetries) {
+            $this->logger->error('Number of retries for SEO inspection exceeded maximum.', [
+                'node_ref' => $command->get('node_ref'),
+                'retries' => $retries,
+                'max_retries' => $maxRetries,
+                'web_result_verdict' => $indexStatusResult ? $indexStatusResult->getVerdict() : 'N/A',
+                'amp_result_verdict' => $ampResult ? $ampResult->getVerdict() : 'N/A',
+                'is_conclusive_for_web' => $isConclusiveForWeb,
+                'is_conclusive_for_amp' => $isConclusiveForAmp,
+            ]);
+            $this->putEvent($command, $pbjx, $node, 'google', $response);
+        } else {
+            $command->addToSet('search_engines', ['google']);
+        }
```

Comment:
@digitalartcrew nit: similar to an early return we often invert if/elses like these when the else is much shorter than the if, like compare

```
if (whatever) {
  // all kind of logic with so many
  // many
  // many
  // many
  // many
  // many
  // many
  // many
  // many
  // many lines of code
} else {
  // just one line
}
```


with


```
if (!whatever) {
  // just one line
} else {
  // all kind of logic with so many
  // many
  // many
  // many
  // many
  // many
  // many
  // many
  // many
  // many lines of code
}
```

in fact with this one you could combine that style with an early return:

```
if ($retries < $maxRetries) {
  $command->addToSet('search_engines', ['google']);
  return $command;
}

// log the warning
// put the event
// return the command

```